### PR TITLE
Improve `PlayRun` reload support

### PR DESCRIPTION
**Pull request by bigdaz**
When a Play application is started with `PlayRun`, Gradle will monitor the input files for changes. Gradle now supports 2 different rebuild behaviours for `PlayRun`: continuous or on-demand.

If Gradle is run with `--continuous` (or `-t`), the application will be rebuilt and reloaded as soon as any file changes are detected. 
Without the `--continuous` flag, Gradle will only rebuild the application on-demand, when an HTTP request is received by the application.

In summary:
- `gradle --continuous runPlay` will start the application, monitor changes, and rebuild/reload the application as soon as a change is detected.
- `gradle runPlay` will start the application, monitor changes, and rebuild/reload the application only when a request is received after a change is detected.

### CODE REVIEW

**Comment by bigdaz** [comment]
@big-guy This is now merged, and I'm happy with the behaviour for 4.2. We now enter continuous build when we detect a reloadable deployment running at the end of the build. It assumes that _any_ deployment left running at the end of the build is 'reloadable', which is true for `PlayApplication` (our only deployment implementation at this stage).

Please take a look and see if there are things you want to improve. Note that I won't be around to respond to feedback, so you'll have to make the changes yourself :).

**Comment by big-guy** [comment]
Thanks for your help on this.

Enjoy your holiday :)
